src/redisraft.c

```diff
@@ -370,6 +370,153 @@ static void handleInfoCommand(RedisRaftCtx *rr,
+} MigrationInfo;
+
+static MigrationInfo * extractMigrationInfo(RedisRaftCtx *rr, RaftRedisCommand *cmd)
+{
+    MigrationInfo * ret = RedisModule_Calloc(1, sizeof(MigrationInfo));
+
+    /* default values if auth isn't specified */
+    strncpy(ret->username, "default", sizeof(ret->username));
+
+    int idx;
+    for (idx = 6; idx < cmd->argc; idx++) {
+        size_t auth_str_len;
+        const char * auth_str = RedisModule_StringPtrLen(cmd->argv[idx], &auth_str_len);
+        if (auth_str_len == 5 && !strncasecmp("AUTH2", auth_str, 5)) {
+            if (idx + 2 > cmd->argc) {
+                LOG_WARNING("couldn't parse username/password");
+                goto fail;
+            }
+
+            size_t username_len;
+            const char * username = RedisModule_StringPtrLen(cmd->argv[idx+1], &username_len);
+            if (username_len > 255) {
+                LOG_WARNING("username is too long: limited to 255 characters");
+                goto fail;
+            }
+            strncpy(ret->username, username, username_len);
+
+            size_t password_len;
+            const char *password = RedisModule_StringPtrLen(cmd->argv[idx+2], &password_len);
+            if (password_len > 255) {
+                LOG_WARNING("password is too long: limited to 255 characters");
+                goto fail;
+            }
+            strncpy(ret->password, password, password_len);
+            break;
+        } else if (auth_str_len == 4 && !strncasecmp("AUTH", auth_str, 4)) {
+            if (idx + 1 > cmd->argc) {
+                LOG_WARNING("couldn't parse password");
+                goto fail;
+            }
+
+            size_t password_len;
+            const char * password = RedisModule_StringPtrLen(cmd->argv[idx+1], &password_len);
+            if (password_len > 255) {
+                LOG_WARNING("username is too long: limited to 255 characters");
+                goto fail;
+            }
+            strncpy(ret->password, password, password_len);
+            break;
+        }
+    }
+
+    return ret;
+
+fail:
+    RedisModule_Free(ret);
+    return NULL;
+}
+
+static RaftReq * cmdToMigrate(RedisRaftCtx *rr, RedisModuleCtx * ctx, RaftRedisCommand *cmd)
+{
+    RaftReq *req = NULL;
+
+    // 0. assert minimum size here
+    if (cmd->argc < 8) {
+        RedisModule_WrongArity(ctx);
+        return req;
+    }
+
+    // 1. Extract migration info and store it in a dict rr->migrate_info
+    MigrationInfo * migrationInfo = extractMigrationInfo(rr, cmd);
+    if (migrationInfo == NULL) {
+        RedisModule_ReplyWithError(ctx, "ERR check logs");
+        goto exit;
+    }
+
+    // 2. assert single key is empty
+    size_t key_len;
+    RedisModule_StringPtrLen(cmd->argv[3], &key_len);
+    if (key_len != 0) {
+        RedisModule_ReplyWithError(ctx, "ERR RedisRaft doesn't support old style single key migration");
+        goto exit;
+    }
+
+    /* find "keys" for parsing out keys */
+    int idx = 6;
```

Comment:
document the reason for the 6 or just start at 0 and do extra string comparisons